pipeline/schema/schema.json

```diff
@@ -243,6 +243,14 @@
           "type": "string",
           "enum": ["fail", "ignore"],
           "default": "fail"
+        },
+        "ports": {
```

Comment:
Maybe we should have two step types (normal & service) based on `detached: true` using [oneOf](https://json-schema.org/understanding-json-schema/reference/combining.html#oneof)